client/castApp/src/dbcb.c

```diff
@@ -71,13 +71,15 @@ static int pushEnv(caster_t *caster)
     }
 
     epicsMutexMustLock(caster->lock);
-    for (i = 0; !ret && i < caster->extra_envs.count; i++) {
-        string_list_t *envptr = (string_list_t *)ellNth(&caster->extra_envs, i);
-        const char *val = getenv(envptr->item_str);
+    ELLNODE *env = ellFirst(&caster->extra_envs);
+    while (!ret && env != NULL) {
+        string_list_t *temp = (string_list_t *)env;
```

Comment:
I don't  love the variable name `temp`; `envptr` was better, and `penvvar` or somethign would be better yet (and more consistent; see how `ellFirst`/`ellNext` is used in EPICS base)